### PR TITLE
MINOR: Small refactor to the Make update_all scripts

### PR DESCRIPTION
**Pull request by IceS2**
### Describe your changes:

- Refactored `update_version.py` script to make it clearer
- Extracted common steps from the release github actions workflows to composite github actions.
- Decoupled the creation of the Release on Github from the execution of the Workflows since they will be triggered programatically from the Release Automation.
- Updated Makefile to use only `RELEASE_VERSION` instea of `RELEASE_VERSION`, `MVN_RELEASE_VERSION` and `PY_RELEASE_VERSION` since we are not doing InPlace releases.

#
### Type of change:

- [x] Improvement

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

**Review by pmbrull — APPROVED** [review]

**Review by pmbrull — APPROVED** [review]

File: .github/workflows/docker-openmetadata-ingestion-base-slim.yml [hunk 1]
```diff
@@ -19,65 +19,31 @@ on:
...
+          is_ingestion: true
 
       - name: Build and push if event is workflow_dispatch and input is checked
         uses: docker/build-push-action@v3
         with:
           context: .
           platforms: linux/amd64,linux/arm64
-          push: ${{ github.event_name == 'release' || github.event_name == 'workflow_dispatch' }}
-          tags: ${{ steps.input_check.outputs.tags }}
+          push: ${{ github.event_name == 'workflow_dispatch' }}
```
Comment (tutte) [inline .github/workflows/docker-openmetadata-ingestion-base-slim.yml:45]: We can hardcode this to `true`

File: .github/workflows/docker-openmetadata-postgres.yml [hunk 2]
```diff
@@ -19,51 +19,29 @@ on:
...
-                echo "tags=openmetadata/postgresql:${{ env.input }},openmetadata/postgresql:latest" >> $GITHUB_OUTPUT
-            elif ${{ github.event_name == 'workflow_dispatch' && inputs.push_latest_tag_to_release }}; then
-                echo "tags=openmetadata/postgresql:${{inputs.tag}},openmetadata/postgresql:latest" >> $GITHUB_OUTPUT
-            else
-                echo "tags=openmetadata/postgresql:${{ inputs.tag }}" >> $GITHUB_OUTPUT
-            fi
+          image: openmetadata/postgresql
+          tag: ${{ inputs.tag }}
+          push_latest: ${{ inputs.push_latest_tag_to_release }}
+          is_ingestion: false
```
Comment (tutte) [inline .github/workflows/docker-openmetadata-postgres.yml:38]: Explicit > implicit, but not needed :)

File: .github/workflows/docker-openmetadata-server.yml [hunk 3]
```diff
@@ -118,52 +85,32 @@ jobs:
...
+          is_ingestion: false
 
       - name: Build and push if event is workflow_dispatch and input is checked
         uses: docker/build-push-action@v3
         with:
           context: .
           platforms: linux/amd64,linux/arm64
-          push: ${{ github.event_name == 'release' || github.event_name == 'workflow_dispatch' }}
-          tags: ${{ steps.input_check.outputs.tags }}
+          push: ${{ github.event_name == 'workflow_dispatch' }}
```
Comment (tutte) [inline .github/workflows/docker-openmetadata-server.yml:112]: Can be hardcoded to true

File: .github/workflows/docker-openmetadata-server.yml [hunk 4]
```diff
@@ -118,52 +85,32 @@ jobs:
...
-                echo "tags=openmetadata/server:${{ env.DOCKER_RELEASE_TAG }},openmetadata/server:latest" >> $GITHUB_OUTPUT
-            elif ${{ github.event_name == 'workflow_dispatch' && inputs.push_latest_tag_to_release }}; then
-                echo "tags=openmetadata/server:${{ env.DOCKER_RELEASE_TAG }},openmetadata/server:latest" >> $GITHUB_OUTPUT
-            else
-                echo "tags=openmetadata/server:${{ env.DOCKER_RELEASE_TAG }}" >> $GITHUB_OUTPUT
-            fi
+          image: openmetadata/server
+          tag: ${{ inputs.DOCKER_RELEASE_TAG }}
+          push_latest: ${{ inputs.push_latest_tag_to_release }}
+          is_ingestion: false
```
Comment (tutte) [inline .github/workflows/docker-openmetadata-server.yml:105]: Same as above

**Review by tutte — APPROVED** [review]
LGTM, much cleaner, easier to read and to expand. Thanks @IceS2

**Review by tutte — APPROVED** [review]

**Pull request merged**